persistence-dse-6.8/src/test/java/io/stargate/db/dse/impl/DsePersistenceIT.java

```diff
@@ -0,0 +1,60 @@
+package io.stargate.db.dse.impl;
+
+import io.stargate.db.Persistence;
+import io.stargate.db.dse.DsePersistenceActivator;
+import io.stargate.it.PersistenceTest;
+import io.stargate.it.storage.ClusterConnectionInfo;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+public class DsePersistenceIT extends PersistenceTest {
+
+  private static File baseDir;
+  private static DsePersistence persistence;
+
+  @BeforeAll
+  public static void createDsePersistence(ClusterConnectionInfo backend) throws IOException {
+    baseDir = Files.createTempDirectory("stargate-dse-test").toFile();
+
+    System.setProperty("stargate.listen_address", "127.0.0.11");
+    System.setProperty("stargate.cluster_name", backend.clusterName());
+    System.setProperty("stargate.datacenter", backend.datacenter());
+    System.setProperty("stargate.rack", backend.rack());
+
+    persistence = new DsePersistence();
+    persistence.initialize(DsePersistenceActivator.makeConfig(baseDir));
+  }
+
+  @AfterAll
+  public static void cleanup() throws IOException {
+    if (persistence != null) {
+      persistence.destroy();
+    }
+
+    FileUtils.deleteDirectory(baseDir);
```

Comment:
If it's a temp directory won't it be cleaned up on it's own?

Reply:
On reboot, yes, but the JVM will not delete it, AFAIK.

We could use `File.deleteOnExit()` but it is not reliable in my experience. I do not think the JVM will do a recursive delete in that case.